docs/pages/machine-id/guides/circleci.mdx

```diff
@@ -0,0 +1,226 @@
+---
+title: Using Machine ID with CircleCI
+description: A tutorial for using Machine ID with CircleCI
+---
+
+<Details
+  title="Version warning"
+  opened={true}
+  scope={["oss", "enterprise"]}
+  scopeOnly={true}
+  min="11.0"
+>
+  Machine ID for CircleCI is available starting from Teleport `v11.1`.
+</Details>
+
+In this guide, you will use Teleport Machine ID to allow a CircleCI job to
+securely connect to a Teleport SSH node without the need for long-lived secrets.
+
+This mitigates the risk of long-lived secrets such as passwords or SSH private
+keys being exfiltrated from your CircleCI organization and provides many of
+the other benefits of Teleport such as auditing and finely-grained access
+control.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+(!docs/pages/includes/tctl.mdx!)
+
+- A node that is a part of the Teleport cluster with
+[Server Access](https://goteleport.com/docs/server-access/introduction/) and a
+user you want to grant access to. In this guide, we will use `my-node` and
+`ci-user` respectively, replace these with values appropriate to your setup.
+- A CircleCI project connected to a Git repository you can push to.
+
+## Step 1/4. Configure CircleCI
+
+In order to configure the rules for which CircleCI workflows will be allowed to
+connect to your Teleport cluster, you must determine the ID of your CircleCI
+organization and create a CircleCI context.
+
+### Find your Organization ID
+
+Open CircleCI and navigate to "Organization settings" from the navbar.
+You should be presented with an interface titled "Overview" with a section
+called "Organization ID". Note this value down and substitute `$ORGANIZATION_ID`
+in configuration examples with this.
+
+### Create a Context
+
+CircleCI has an organization level concept called
+["contexts"](https://circleci.com/docs/contexts/). Contexts allow an
+organization to configure a series of secrets that should be exposed to a
+workflow job, and can control which actors are allowed to trigger jobs
+associated with a context.
```

Comment:
```suggestion
CircleCI has an organization-level concept called
**contexts**, which allow you to configure a series of secrets that should be exposed to a
workflow job. You can configure CircleCI to control which actors are allowed to trigger jobs
associated with a context.
```
Since reading more information about contexts isn't necessary to complete this guide, I've suggested removing the link. Should we add the link to the "Further steps" section?

Reply:
We already link to the general CircleCI docs, I think it'll be fine.